http/security/helper.go

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2019 by PACE Telematics GmbH. All rights reserved.
+// Created at 2019/10/30 by Charlotte Pröller
+
+package security
+
+import (
+	"context"
+	"net/http"
+	"strings"
+)
+
+type Token interface {
+	GetValue() string
+}
+
+type bearerToken struct {
+	value string
+}
+
+type ctxkey string
+
+const HeaderPrefix = "Bearer "
+
+var tokenKey = ctxkey("Token")
+
+// GetValue returns the be
+func (b *bearerToken) GetValue() string {
+	return b.value
+}
+
+// GetBearerTokenFromHeader get the bearer Token from the header of the request
+// success: returns the bearer token without the Prefix
+// error: write the error in the response and return a empty bearer
+func GetBearerTokenFromHeader(r *http.Request, w http.ResponseWriter, headerName string) string {
```

Comment:
I'm confused about all these methods: `GetBearerTokenFromHeader`, `ContextWithTokenKey`, `TokenFromContext`, `WithBearerToken` and `BearerToken`.

There is no harmony in this. `WithBearerToken` and `BearerToken` are very symmetrical, but their naming is totally different from the very similar functions `ContextWithTokenKey` and `TokenFromContext`. `GetBearerTokenFromHeader` has no equivalent setter.

When should I use `ContextWithTokenKey` and when `WithBearerToken`? It seems redundant to me having both functions.

`GetBearerTokenFromHeader` is not a getter, it might actually call `http.Error(w, "Unauthorized", http.StatusUnauthorized)`. But the name suggests that it is only a getter.

I would suggest that you reduce the number of functions, for example:
```go
func WithToken(context.Context, Token) context.Context {}
func GetToken(context.Context) Token {}
```

I'm not sure what to do about the `http.Error(w, "Unauthorized", http.StatusUnauthorized)`. It seems so be misplaced in `GetBearerTokenFromHeader` but I'm not sure where it belongs. Making it part of the `Authenticator`'s `Authorize` method could work.